### ISSUE TITLE
NG Compactor: Split compacted data into 2 files

### ISSUE THREAD

**Issue by NGA-TRAN** [influxdata/influxdb_iox#3999]
# Rationale

This ticket implements step 6.b.ii of [Find and Compact files](https://docs.google.com/document/d/1lfILhw5ckzTlXFOk7VXKQOiz3TvDDlx-tdCgQ0YiG8E/edit#heading=h.y3uqgzkpk0vx). The input is the compacted data and the output is 2 files split based on the given split_rules. Something like

```rust
fn split compact(
    CompactedData, 
    split_rules,  // needs to decide what the rules are
) -> Result<out1, out2)>

```

**Comment by alamb** [comment]
First of all, when the partition is "cold" (aka not getting writes anymore) I think we should compact the entire thing into a single file (subject to the max file size limit)

For "hot" partitions that are still getting writes, I agree with @NGA-TRAN  that approach 1 sounds good 👍  because:
1. It seems like it is less work (we can reuse the existing query plans)
2. It wouldn't require much rework if we eventually decide we need something like approach 2

I think the whole point of splitting when compacting a "hot" partition (that is still getting data) is to make sure the larger part doesn't overlap with new data that will come in. A static 10/90 split based on timestamp seems reasonable at first, though of course we will be able to find pathological counter examples for any policy that is chosen,

**Comment by NGA-TRAN** [comment]
@alamb Good point regarding cold and hot partition. Let assume we do this for hot partition. I have added 2 tasks in the `Improvement` section of parent ticket #3904 to handle cold partition and hierarchical partition if needed
